### PR TITLE
feat: adds new ci jobs for Secrets Store Sync Controller

### PR DESCRIPTION
**Pull request by nilekhc**
/cc @aramase

### CODE REVIEW

File: config/jobs/kubernetes-sigs/secrets-store-sync-controller/secrets-store-sync-controller-config.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      timeout: 10m
+    always_run: true
+    path_alias: sigs.k8s.io/secrets-store-sync-controller
+    branches:
+    - ^main$
+    - ^release-*
+    labels:
+    spec:
+      containers:
+      - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20231113-7213ea5323-1.25
```
Comment (googs1025) [inline config/jobs/kubernetes-sigs/secrets-store-sync-controller/secrets-store-sync-controller-config.yaml:16]: hi! The new image has been released, maybe we should use the latest image
`gcr.io/k8s-staging-test-infra/kubekins-e2e:v20240515-17c6d50e24-master`

Comment (aramase) [inline config/jobs/kubernetes-sigs/secrets-store-sync-controller/secrets-store-sync-controller-config.yaml:16]: No need to pin this to `-1.25`. Let's use the latest image `kubekins` image.

File: config/jobs/kubernetes-sigs/secrets-store-sync-controller/secrets-store-sync-controller-config.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      timeout: 10m
+    always_run: true
+    path_alias: sigs.k8s.io/secrets-store-csi-driver
+    branches:
+    - ^main$
+    - ^release-*
+    labels:
+    spec:
+      containers:
+      - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20231113-7213ea5323-1.25
```
Comment (aramase) [inline config/jobs/kubernetes-sigs/secrets-store-sync-controller/secrets-store-sync-controller-config.yaml:49]: This image should be updated to latest `gcr.io/k8s-staging-test-infra/kubekins-e2e:v20240515-17c6d50e24-master`

File: config/jobs/kubernetes-sigs/secrets-store-sync-controller/secrets-store-sync-controller-config.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+          privileged: true
+        resources:
+          requests:
+            cpu: "4"
+            memory: "8Gi"
+          limits:
+            cpu: "4"
+            memory: "8Gi"
+    annotations:
+      testgrid-dashboards: sig-auth-secrets-store-sync-controller
```
Comment (aramase) [inline config/jobs/kubernetes-sigs/secrets-store-sync-controller/secrets-store-sync-controller-config.yaml:65]: This dashboard doesn't exist, I would recommend just using `sig-auth-secrets-store-csi-driver` dashboard for now

File: config/jobs/kubernetes-sigs/secrets-store-sync-controller/secrets-store-sync-controller-config.yaml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
```
Comment (aramase) [inline config/jobs/kubernetes-sigs/secrets-store-sync-controller/secrets-store-sync-controller-config.yaml:32]: This dashboard doesn't exist, I would recommend just using `sig-auth-secrets-store-csi-driver` dashboard for now so we can make progress. 

We can create `sig-auth-secrets-store-sync-controller` later and move the jobs.

**Review by aramase — CHANGES_REQUESTED** [review]

**Review by aramase — COMMENTED** [review]
/lgtm
/approve

**Comment by aramase** [comment]
/assign @dims 
for approval

**Review by aramase — COMMENTED** [review]
/lgtm

**Pull request merged**